### PR TITLE
Questions after review PT

### PR DESCRIPTION
**Pull request by SamirBoulil**
Some questions raised after a full review of the PT revamp code (back/front).

☝️ Please let me know if you agree with the comments and also if think this is something we should do in the near future to close this topic.
If that's the case, I'll create a dedicated JIRA task to do them all.

### CODE REVIEW

File: src/Akeneo/UserManagement/Component/Model/Role.php [hunk 1]
```diff
@@ -14,7 +14,7 @@ class Role implements RoleInterface
     protected ?int $id = null;
     protected ?string $role = null;
     protected ?string $label = null;
-    protected string $type = self::TYPE_DEFAULT;
+    protected ?string $type = self::TYPE_DEFAULT;
```
Comment (SamirBoulil) [inline src/Akeneo/UserManagement/Component/Model/Role.php:17]: oups :) that's out of the scope :)

File: src/Akeneo/Platform/Job/back/Infrastructure/Query/SearchJobExecution.php [hunk 2]
```diff
@@ -28,6 +34,11 @@ public function __construct(Connection $connection)
 
     public function search(SearchJobExecutionQuery $query): array
     {
+        // I'd split this public into 2 private for clarity,
+        // the body of this function becomes:
+        // $sql = $this->buildSqlQuery($query);
+        // return $this->fetchJobExecutionRow($sql);
```
Comment (pierrejolly) [inline src/Akeneo/Platform/Job/back/Infrastructure/Query/SearchJobExecution.php:40]: 👍

File: src/Akeneo/Platform/Job/back/Infrastructure/Symfony/Resources/config/test/services.yml [hunk 3]
```diff
@@ -1,4 +1,5 @@
 services:
+    # Is this service definition file loaded in the prod kernel ?
```
Comment (ghoussard) [inline src/Akeneo/Platform/Job/back/Infrastructure/Symfony/Resources/config/test/services.yml:2]: Nop, it is loaded only in test kernel (cf src/Akeneo/Platform/Job/back/Infrastructure/Symfony/DependencyInjection/AkeneoJobExtension.php line 25)

[the remark this answers is not in the document]
Comment (SamirBoulil) [reply src/Akeneo/Platform/Job/back/Infrastructure/Symfony/Resources/config/test/services.yml:2]: Thanks :)

Comment (SamirBoulil) [reply src/Akeneo/Platform/Job/back/Infrastructure/Symfony/Resources/config/test/services.yml:2]: Thanks !

File: src/Akeneo/Platform/Job/front/process-tracker/src/feature/pages/JobExecutionList.tsx [hunk 4]
```diff
@@ -51,11 +51,13 @@ const JobExecutionList = () => {
       <PageContent>
         {jobExecutionTable && (
           <Search sticky={0} placeholder="TODO RAC-938" searchValue="" onSearchChange={handleSearchChange}>
+            {/*⬆️ TODO RAC-938 ?*/}
```
Comment (ValentinMumble) [inline src/Akeneo/Platform/Job/front/process-tracker/src/feature/pages/JobExecutionList.tsx:54]: Yes it's merged in the feature branch

File: src/Akeneo/Platform/Job/back/Application/SearchJobExecution/FindJobUsersQuery.php [hunk 5]
```diff
@@ -3,6 +3,14 @@
 namespace Akeneo\Platform\Job\Application\SearchJobExecution;
 
 /**
+ * Maybe this is just another read usecase ? How about ?
```
Comment (ghoussard) [inline src/Akeneo/Platform/Job/back/Application/SearchJobExecution/FindJobUsersQuery.php:6]: 👍🏻

File: src/Akeneo/Platform/Job/back/Infrastructure/Controller/IndexAction.php [hunk 6]
```diff
@@ -42,23 +42,13 @@ public function __invoke(Request $request): Response
...
-        $sort = $request->get('sort');
-        $searchJobExecutionQuery->sortColumn = $sort['column'] ?? 'started_at';
-        $searchJobExecutionQuery->sortDirection = $sort['direction'] ?? 'DESC';
-        $searchJobExecutionQuery->type = $request->get('type', []);
-        $searchJobExecutionQuery->user = $user;
-        $searchJobExecutionQuery->status = $request->get('status', []);
-        $searchJobExecutionQuery->search = $request->get('search', '');
-        $searchJobExecutionQuery->code = $request->get('code', []);
-
+        $searchJobExecutionQuery = $this->createSearchQuery($request, $user);
```
Comment (ghoussard) [inline src/Akeneo/Platform/Job/back/Infrastructure/Controller/IndexAction.php:51]: I'm refactoring that in RAC-1011.
The createSearchQuery will contains validation also.

File: src/Akeneo/Platform/Job/back/Infrastructure/Controller/GetJobUsersAction.php [hunk 7]
```diff
@@ -41,19 +41,24 @@ public function __invoke(Request $request): Response
...
+
         $findJobUsersQuery = new FindJobUsersQuery();
         $findJobUsersQuery->search = (string) $request->get('search', '');
 
         $jobUsers = $this->findJobUsers->search($findJobUsersQuery);
 
         return new JsonResponse($jobUsers);
     }
+
+    private function denyAccessUnlessAclIsGranted()
```
Comment (ghoussard) [inline src/Akeneo/Platform/Job/back/Infrastructure/Controller/GetJobUsersAction.php:58]: 👍🏻

File: src/Akeneo/Platform/Job/back/Infrastructure/Query/FindJobUsers.php [hunk 8]
```diff
@@ -23,14 +23,21 @@ public function __construct(Connection $connection)
     }
 
     public function search(FindJobUsersQuery $query): array
+    {
```
Comment (ghoussard) [inline src/Akeneo/Platform/Job/back/Infrastructure/Query/FindJobUsers.php:26]: 👍🏻

File: src/Akeneo/Platform/Job/back/Infrastructure/Controller/GetJobUsersAction.php (hunk 7 above)
Comment (ValentinMumble) [inline src/Akeneo/Platform/Job/back/Infrastructure/Controller/GetJobUsersAction.php:58]:
```suggestion
    private function denyAccessUnlessAclIsGranted(): void
```

File: src/Akeneo/Platform/Job/back/Infrastructure/Query/SearchJobExecution.php [hunk 9]
```diff
@@ -13,6 +13,12 @@
 use Doctrine\DBAL\Types\Types;
 
 /**
+ * TODO: Should we extract this class into multiple classes ?
+ *  - To manage filters: instead of managing both at the same time in private classes buildQueryParams and buildQueryParamsTypes ?
+ *      - TypeFilter class: responsible for handling type filter
+ *      - StatusFilter class: responsible for handling the Status filtering
+ *  - To manage JobExecutionRow hydration in a dedicated class too
+ *
```
Comment (ValentinMumble) [inline src/Akeneo/Platform/Job/back/Infrastructure/Query/SearchJobExecution.php:21]: 👍

File: src/Akeneo/Platform/Job/front/process-tracker/src/feature/pages/JobExecutionList.test.tsx [hunk 10]
```diff
@@ -187,3 +187,11 @@ test('it can change page', () => {
   expect(screen.queryByText('Export job')).not.toBeInTheDocument();
   expect(screen.getByText('Mass edit')).toBeInTheDocument();
 });
+
+// missing test when pagination is changed
```
Comment (ValentinMumble) [inline src/Akeneo/Platform/Job/front/process-tracker/src/feature/pages/JobExecutionList.test.tsx:191]: https://github.com/akeneo/pim-community-dev/blob/20ceeb9ab9964e6653e2527bd08fa5ecf6932248/src/Akeneo/Platform/Job/front/process-tracker/src/feature/pages/JobExecutionList.test.tsx#L181

File: src/Akeneo/Platform/Job/front/process-tracker/src/feature/pages/JobExecutionList.test.tsx [hunk 11]
```diff
@@ -187,3 +187,11 @@ test('it can change page', () => {
...
+// missing test when type filter is changed
```
Comment (ValentinMumble) [inline src/Akeneo/Platform/Job/front/process-tracker/src/feature/pages/JobExecutionList.test.tsx:192]: https://github.com/akeneo/pim-community-dev/blob/20ceeb9ab9964e6653e2527bd08fa5ecf6932248/src/Akeneo/Platform/Job/front/process-tracker/src/feature/pages/JobExecutionList.test.tsx#L149 ?

**Review by ValentinMumble — APPROVED** [review]

File: src/Akeneo/Platform/Job/front/process-tracker/src/feature/pages/JobExecutionList.test.tsx (hunk 10 above)
Comment (SamirBoulil) [reply src/Akeneo/Platform/Job/front/process-tracker/src/feature/pages/JobExecutionList.test.tsx:191]: Those comments are from my review on master 😅

File: src/Akeneo/Platform/Job/back/Infrastructure/Controller/GetJobUsersAction.php (hunk 7 above)
Comment (SamirBoulil) [reply src/Akeneo/Platform/Job/back/Infrastructure/Controller/GetJobUsersAction.php:58]: done

File: src/Akeneo/Platform/Job/back/Application/SearchJobExecution/FindJobUsersQuery.php (hunk 5 above)
Comment (StevenVAIDIE) [reply src/Akeneo/Platform/Job/back/Application/SearchJobExecution/FindJobUsersQuery.php:6]: Yes it's very simple but it's an use case for for me it should be in Application layer.

For me we should move it into Application/SearchUsers/SearchUsersQuery

Comment (SamirBoulil) [reply src/Akeneo/Platform/Job/back/Application/SearchJobExecution/FindJobUsersQuery.php:6]: Even if it's simple, it's still a usecase it should appear as clearly as possible as is no ?

File: src/Akeneo/Platform/Job/back/Infrastructure/Controller/GetJobUsersAction.php [hunk 12]
```diff
@@ -41,19 +41,24 @@ public function __invoke(Request $request): Response
...
 
-        if (!$this->securityFacade->isGranted('pim_enrich_job_tracker_index')) {
-            throw new AccessDeniedHttpException('Access forbidden. You are not allowed to list job users.');
-        }
-
         if (!$this->securityFacade->isGranted('pim_enrich_job_tracker_view_all_jobs')) {
             return new JsonResponse([$this->security->getUser()->getUserIdentifier()]);
         }
 
+        $this->denyAccessUnlessAclIsGranted();
```
Comment (StevenVAIDIE) [inline src/Akeneo/Platform/Job/back/Infrastructure/Controller/GetJobUsersAction.php:48]: For me it should be checked before the line 44, we should deny even if the user have the ACL pim_enrich_job_tracker_view_all_jobs

File: src/Akeneo/Platform/Job/back/Infrastructure/Controller/IndexAction.php [hunk 13]
```diff
@@ -42,23 +42,13 @@ public function __invoke(Request $request): Response
         }
```
Comment (StevenVAIDIE) [inline src/Akeneo/Platform/Job/back/Infrastructure/Controller/IndexAction.php:43]: This class should be renamed for me => ProcessTrackerAction or JobExecutionAction

File: src/Akeneo/Platform/Job/back/Infrastructure/Controller/IndexAction.php (hunk 6 above)
Comment (SamirBoulil) [reply src/Akeneo/Platform/Job/back/Infrastructure/Controller/IndexAction.php:51]: I reverted

File: src/Akeneo/Platform/Job/back/Application/SearchJobExecution/FindJobUsersQuery.php (hunk 5 above)
Comment (StevenVAIDIE) [reply src/Akeneo/Platform/Job/back/Application/SearchJobExecution/FindJobUsersQuery.php:6]: Same for FindJobTypesInterface

**Review by StevenVAIDIE — APPROVED** [review]

File: src/Akeneo/Platform/Job/back/Infrastructure/Query/FindJobUsers.php [hunk 14]
```diff
@@ -23,14 +23,21 @@ public function __construct(Connection $connection)
     }
 
     public function search(FindJobUsersQuery $query): array
```
[the remark this answers is not in the document]
Comment (pierrejolly) [reply src/Akeneo/Platform/Job/back/Infrastructure/Query/FindJobUsers.php:25]: The class name is already FindJobUsers I'm not sure it's useful

[the remark this answers is not in the document]
Comment (StevenVAIDIE) [reply src/Akeneo/Platform/Job/back/Infrastructure/Query/FindJobUsers.php:25]: I wanted to say `This class should be renamed to SearchJobUsers ?` sorry

File: src/Akeneo/Platform/Job/back/Infrastructure/Controller/IndexAction.php (hunk 13 above)
Comment (SamirBoulil) [reply src/Akeneo/Platform/Job/back/Infrastructure/Controller/IndexAction.php:33]: done in https://github.com/akeneo/pim-community-dev/pull/15861

**Pull request closed without merging**